src/test/java/org/numenta/nupic/encoders/RandomDistributedScalarEncoderTest.java

```diff
@@ -0,0 +1,438 @@
+		rdse.bucketMap.put(midIdx-2, getRangeAsList(3, 8));
+		rdse.bucketMap.put(midIdx-1, getRangeAsList(4, 9));
+		rdse.bucketMap.put(midIdx, getRangeAsList(5, 10));
+		rdse.bucketMap.put(midIdx+1, getRangeAsList(6, 11));
+		rdse.bucketMap.put(midIdx+2, getRangeAsList(7, 12));
+		rdse.bucketMap.put(midIdx+3, getRangeAsList(8, 13));
+		rdse.minIndex = midIdx-3;
+		rdse.maxIndex = midIdx+3;
+		
+		try {
+			assertTrue("overlapOK didn't work", rdse.overlapOK(midIdx, midIdx-1));
+			assertTrue("overlapOK didn't work", rdse.overlapOK(midIdx-2, midIdx+3));
+			assertFalse("overlapOK didn't work", rdse.overlapOK(midIdx-3, midIdx-1));
+			
+			assertTrue("overlapOK didn't work for far values", rdse.overlapOK(100, 50, 0));
+			assertTrue("overlapOK didn't work for far values", rdse.overlapOK(100, 50, rdse.getMaxOverlap()));
+			assertFalse("overlapOK didn't work for far values", rdse.overlapOK(100, 50, rdse.getMaxOverlap()+1));
+			assertTrue("overlapOK didn't work for far values", rdse.overlapOK(50, 50, 5));
+			assertTrue("overlapOK didn't work for far values", rdse.overlapOK(48, 50, 3));
+			assertTrue("overlapOK didn't work for far values", rdse.overlapOK(46, 50, 1));
+			assertTrue("overlapOK didn't work for far values", rdse.overlapOK(45, 50, rdse.getMaxOverlap()));
+			assertFalse("overlapOK didn't work for far values", rdse.overlapOK(48, 50, 4));
+			assertFalse("overlapOK didn't work for far values", rdse.overlapOK(48, 50, 2));
+			assertFalse("overlapOK didn't work for far values", rdse.overlapOK(46, 50, 2));
+			assertFalse("overlapOK didn't work for far values", rdse.overlapOK(50, 50, 6));
+		} catch (IllegalStateException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+	}
+	
+	@Test
+	public void testCountOverlap()
+	{
+		builder = RandomDistributedScalarEncoder.builder()
+				.name("enc")
+				.resolution(1)
+				.n(500);
+		rdse = builder.build();
+		
+		int[] r1 = new int[]{1, 2, 3, 4, 5, 6};
+		int[] r2 = new int[]{1, 2, 3, 4, 5, 6};
+		assertEquals("countOverlap result is incorrect", 6, rdse.countOverlap(r1, r2));
+		
+		r1 = new int[]{1, 2, 3, 4, 5, 6};
+		r2 = new int[]{1, 2, 3, 4, 5, 7};
+		assertEquals("countOverlap result is incorrect", 5, rdse.countOverlap(r1, r2));
+		
+		r1 = new int[]{1, 2, 3, 4, 5, 6};
+		r2 = new int[]{6, 5, 4, 3, 2, 1};
+		assertEquals("countOverlap result is incorrect", 6, rdse.countOverlap(r1, r2));
+		
+		r1 = new int[]{1, 2, 8, 4, 5, 6};
+		r2 = new int[]{1, 2, 3, 4, 9, 6};
+		assertEquals("countOverlap result is incorrect", 4, rdse.countOverlap(r1, r2));
+		
+		r1 = new int[]{1, 2, 3, 4, 5, 6};
+		r2 = new int[]{1, 2, 3};
+		assertEquals("countOverlap result is incorrect", 3, rdse.countOverlap(r1, r2));
+		
+		r1 = new int[]{7, 8, 9, 10, 11, 12};
+		r2 = new int[]{1, 2, 3, 4, 5, 6};
+		assertEquals("countOverlap result is incorrect", 0, rdse.countOverlap(r1, r2));
+	}
+	
+	@Test
+	public void testVerbosity()
+	{
+		
+	}
+	
+	@Test
```

Comment:
Add ``@Ignore`` to tests that aren't implemented. Or add ``fail("useful message")`` to the test body, but I think the latter will block the pull request, right?